src/cluster.c

```diff
@@ -1131,8 +1131,9 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
         margv = ms->commands[i].argv;
 
         /* Only valid for sharded pubsub as regular pubsub can operate on any node and bypasses this layer. */
-        if (!pubsubshard_included &&
-            doesCommandHaveChannelsWithFlags(mcmd, CMD_CHANNEL_PUBLISH | CMD_CHANNEL_SUBSCRIBE))
+        if (!pubsubshard_included && (mcmd->proc == ssubscribeCommand ||
+                                      mcmd->proc == sunsubscribeCommand ||
+                                      mcmd->proc == spublishCommand))
```

Comment:
why not `doesCommandHaveChannelsWithFlags(mcmd, CMD_CHANNEL_PUBLISH | CMD_CHANNEL_SUBSCRIBE | CMD_CHANNEL_UNSUBSCRIBE))`

Reply:
It's unnecessary to iterate commands_with_channels, and even though the regular pubsub doesn't go through this, I still think it's clearer to write it this way.

Reply:
it makes this specific piece of code clearer, but i don't think it's overall better when considering the entire code base.
i.e. if we one day add more commands that are similar to these, we'll update the table, and all the users of that function will implicitly work. i'd rather concentrate the logic on one place, and if possible avoid any explicit `if (cmd->proc==` and instead add flags to the commands, so that it's always possible to add more commands that behave the same.

if you'll look at the code before redis 7.0, there were a ton of places with such `if`s, sometimes matching 10 different commands. we attempted to change as many of these as possible to flags (this way, modules can declare commands with these flags too).
in the few places where flags didn't make too much sense, we still tried to move all the matching to one place (e.g. the `commands_with_channels` table).